### ISSUE TITLE
`process.env` is undefined

### ISSUE THREAD

**Issue by KaelWD** [nuxt/framework#2797]
### Environment

```
Nuxt CLI v3.0.0-27375427.d07d572
RootDir: /home/kael/Documents/vuetifyjs/test-projects/nuxt3/nuxt3-app
Nuxt project info:
------------------------------
- Operating System: `Linux`
- Node Version:     `v16.13.2`
- Nuxt Version:     `3.0.0-27375427.d07d572`
- Package Manager:  `yarn@1.22.5`
- Bundler:          `Vite`
- User Config:      `css`, `build`, `vite`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------
```

### Reproduction

Create a plugin with

```js
console.log(typeof process)
console.log(process.env.NODE_ENV)
```

and look at the browser console. 

### Describe the bug

`process` is `{ dev: true, server: false, client: true }`, should be `undefined`
`process.env.NODE_ENV` throws `TypeError: Cannot read properties of undefined (reading 'NODE_ENV')`, should be `'development'`

### Additional context

This works with just vite, idk what you're doing to disable that. It looks like this is also set up correctly for builds with `@rollup/plugin-replace`, so it's only a problem during dev.

**Comment by danielroe** [comment]
Would you provide a reproduction? 🙏 z

**Comment by KaelWD** [comment]
Not sure what happened here, it seems to be working now.

**Comment by danielroe** [comment]
Magic ✨

**Comment by Saul-Mirone** [comment]
This seems appear again. The `process` in browser doesn't have a `env` property that breaks some packages.

**Comment by Saul-Mirone** [comment]
> I guess vite only supports env using `import.meta.env`. Related enhahncenemnt: #1502

Yeah, would you consider giving it some workaround to make packages that depend on `process.env` work? For now, packages like [kleur](https://github.com/lukeed/kleur/blob/master/index.mjs#L5) will be broken.

**Comment by pi0** [comment]
I'm certain we can! Do you mind make a small sandbox with kleur and nuxt 3 I can fork?

**Comment by Saul-Mirone** [comment]
> 

Sure, here it is: https://stackblitz.com/edit/github-fkzpox?file=app.vue

**Comment by Extarys** [comment]
I started having this issue again, last time was in November.

If I can do something to help find the cause of this issue...

**Comment by ennioVisco** [comment]
> https://stackblitz.com/edit/github-fkzpox?file=app.vue

@Saul-Mirone the sandbox link that you provided is not working... Could you please update it? 
Please also update nuxt to rc9!

**Comment by Saul-Mirone** [comment]
> > https://stackblitz.com/edit/github-fkzpox?file=app.vue
> 
> @Saul-Mirone the sandbox link that you provided is not working... Could you please update it? Please also update nuxt to rc9!

I can't reproduce it in rc9 now. 👍
